### DISCUSSION TITLE
Intel Mac 13.3.1: Closer to working

### DISCUSSION THREAD

**Discussion by shawncaple** [vladmandic/sdnext#200] [General]
### Issue Description

Just wanted to give an update of where the current version lies with regards to a Mac Setup. I know this fork hasn't been optimized for a Mac system. This version got further along than previous ones. I hope this can help sort out issues for future versions or anyone that could build a fix for MacOS.

16:02:39-866933 INFO     Python 3.10.9 on Darwin                                
16:02:39-931977 INFO     Running setup                                          
16:02:39-944076 INFO     Version: 133e2855 Mon Apr 17 10:03:33 2023 -0400       
16:02:40-420503 INFO     Latest published version:                              
                         09a1480233539d049244cb722706a10b53837bb9               
                         2023-04-17T19:52:26Z                                   
16:02:40-421889 INFO     Updating Wiki                                          
16:02:42-393291 INFO     Installing requirements                                
16:02:42-582308 WARNING  Package wrong version: timm 0.4.12 required 0.6.13     
16:02:45-425521 INFO     Installing packages                                    
16:02:45-957611 ERROR    Error running pip with args: install --upgrade         
                         --no-deps xformers==0.0.17                             
16:02:45-959701 INFO     Installing repositories                                
16:02:46-020451 INFO     Installing submodules                                  
16:02:46-884720 INFO     Updating submodules                                    
16:02:52-455011 INFO     Extensions enabled: ['SwinIR',                         
                         'sd-extension-steps-animation',                        
                         'clip-interrogator-ext',                               
                         'sd-extension-aesthetic-scorer',                       
                         'sd-dynamic-thresholding', 'prompt-bracket-checker',   
                         'sd-webui-controlnet', 'ScuNET',                       
                         'stable-diffusion-webui-rembg',                        
                         'sd-webui-model-converter', 'Lora',                    
                         'sd-extension-system-info',                            
                         'stable-diffusion-webui-images-browser', 'LDSR',       
                         'seed_travel',                                         
                         'multidiffusion-upscaler-for-automatic1111',           
                         'a1111-sd-webui-lycoris']                              
16:03:11-607463 INFO     Extensions enabled: []                                 
16:03:11-608800 WARNING  Setup complete with errors (1)                         
16:03:11-609534 INFO     Setting environment tuning                             
16:03:13-171947 ERROR    Error running pip with args: install --upgrade torch   
                         torchaudio torchvision --index-url                     
                         https://download.pytorch.org/whl/cu118                 
16:03:14-126124 INFO     Torch 2.0.0                                            
16:03:14-127117 WARNING  Torch repoorts CUDA not available                      
16:03:14-127786 INFO     Server arguments: ['--no-half', '--no-half-vae']       
No module 'xformers'. Proceeding without it.
Available models: /Users/xxxx/automatic/models/Stable-diffusion 1
OMP: Error #15: Initializing libomp.dylib, but found libiomp5.dylib already initialized.
OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://openmp.llvm.org/
zsh: abort      ./webui.sh

### Version Platform Description

iMac 
Ventura 13.3.1
3.6 GHz 10-Core Intel Core i9
AMD Radeon Pro 5700 XT 16 GB
128 GB 2667 MHz DDR4

**Comment by vladmandic** [comment]
Thanks for the update. I'd love to collect info from Apple uses so I can bake it all in and just-work.
But until then I think, this is better suited for discussions?

**Reply by shawncaple** [reply]
Got it. I'll continue to check in as it progresses.

**Comment by shawncaple** [comment]
Mac user update for Intel Mac: as of now, this version installed, except I had to add the v1-5-pruned-emaonly.ckpt to the Stable Diffusion folder manually. Then in opened the GUI. I typed in "Cat". Hit "Generate" and got his:

Initializing ━━━━━━━━━━━━━━━━━━━━━━   0% -:--:-- 0:00:00loc("mps_add"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/97f6331a-ba75-11ed-a4bc-863efbbaf80d/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":228:0)): error: input types 'tensor<2x1280xf32>' and 'tensor<1280xf16>' are not broadcast compatible
LLVM ERROR: Failed to infer result type(s).
zsh: abort      ./webui.sh

**Comment by itsastroa** [comment]
i got the same issue 23:46:50-595796 INFO     Python 3.10.9 on Windows
23:46:51-577800 INFO     Version: bfe62127 Thu Apr 27 15:47:13 2023 -0400
23:46:52-401954 INFO     Setting environment tuning
23:46:52-407654 INFO     nVidia toolkit detected
23:46:57-944227 INFO     Torch 2.0.0+cu118
23:46:59-163856 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
23:46:59-167894 INFO     Torch detected GPU: NVIDIA GeForce RTX 2060 VRAM 6144 Arch (7, 5) Cores 30
23:46:59-171036 INFO     Installing package: tensorflow==2.12.0
23:48:28-127909 INFO     Verifying requirements
23:48:28-169123 WARNING  Package wrong version: timm 0.4.12 required 0.6.13
23:48:28-172122 INFO     Installing package: timm==0.6.13
23:48:38-129805 INFO     Running setup
23:48:38-133813 INFO     Installing packages
23:48:38-135814 INFO     Installing repositories
23:48:38-578570 INFO     Installing submodules
23:48:44-137350 INFO     Updating submodules

**Comment by vladmandic** [comment]
Sorry, but nVidia 2060 cannot be same issue as Apple platform for which issue was opened for?
